demo/src/screens/MainScreen.js

```diff
@@ -50,15 +53,39 @@ class MainScreen extends Component {
     super(props);
 
     const data = props.navigationData || navigationData;
-
     this.state = {
       currentPage: 0,
-      filteredNavigationData: data
+      filteredNavigationData: data,
+      chipsLabels: _.map(data, section => section.title),
+      sectionsData: _.map(data, section => ({title: section.title, data: section.screens})),
+      selectedSection: 0,
+      faderStart: false,
+      faderEnd: true
     };
 
     Navigation.events().bindComponent(this);
   }
 
+  sectionListRef = React.createRef();
+  scrollViewRef = React.createRef();
+
+  onPressFlag = false;
+  onUserScroll = false;
+
+  componentDidUpdate(prevState) {
+    const {selectedSection} = this.state;
+    if (prevState.selectedSection !== selectedSection) {
+      if (this.onPressFlag) {
+        this.scrollToSection(selectedSection);
+        this.scrollChipsSection(selectedSection);
+      }
+      if (this.onUserScroll) {
+        this.scrollChipsSection(selectedSection);
```

Comment:
Is it possible that both flag will be true? 
If it's possible, they you might get a use case where you call `this.scrollChipsSection` twice

Reply:
No there is no case that both of them are true.